controllers/replace_misconfigured_pods.go

```diff
@@ -221,9 +221,61 @@ func instanceNeedsRemoval(cluster *fdbtypes.FoundationDBCluster, instance FdbIns
 		}
 	}
 
+	if cluster.Spec.ReplaceInstancesWhenResourcesChange != nil && *cluster.Spec.ReplaceInstancesWhenResourcesChange {
+		desiredSpec, err := GetPodSpec(cluster, instance.GetProcessClass(), idNum)
+		if err != nil {
+			return false, err
+		}
+
+		for idx, container := range desiredSpec.Containers {
+			if equality.Semantic.DeepEqual(container.Resources, instance.Pod.Spec.Containers[idx].Resources) {
+				continue
+			}
+
+			if resourcesNeedsReplacement(container.Resources, instance.Pod.Spec.Containers[idx].Resources) {
```

Comment:
Now that I merged it I had another idea. It probably makes sense to just compare the sum of the requests (e.g. in the current implementation we probably also don't handle the case for additional containers).